src/ButtonGroup/__tests__/ButtonGroup.test.js

```diff
@@ -1,143 +1,131 @@
+    expect(onButton).not.toHaveAttribute('data-appearance', 'secondary');
+    expect(offButton).not.toHaveAttribute('data-isactive', 'true');
+    expect(offButton).not.toHaveAttribute('data-appearance', 'secondary');
   });
 
-  it('should respond to a click handler', () => {
+  test('should respond to a click handler', () => {
     const spy = jest.fn();
-    const render = mount(
-      <ButtonGroup theme={Theme} defaultActiveButton="ON" onChange={spy}>
-        <Button theme={Theme} name="ON" appearance="secondary">
+    const { getByText } = render(
+      <ButtonGroup defaultActiveButton="ON" onChange={spy}>
+        <Button name="ON" appearance="secondary">
           ON
         </Button>
-        <Button theme={Theme} name="OFF" appearance="secondary">
+        <Button name="OFF" appearance="secondary">
           OFF
         </Button>
       </ButtonGroup>,
     );
 
-    render
-      .find(Button)
-      .last()
-      .simulate('click');
+    const offButton = getByText('OFF');
+
+    fireEvent.click(offButton);
 
     expect(spy).toHaveBeenCalledTimes(1);
   });
 
-  it('should call the onclick handler on child button', () => {
+  test('should call the onclick handler on child button', () => {
     const spy = jest.fn();
     const spy2 = jest.fn();
-    const render = mount(
-      <ButtonGroup theme={Theme} defaultActiveButton="ON" onChange={spy}>
-        <Button theme={Theme} name="ON" appearance="secondary">
+    const { getByText } = render(
+      <ButtonGroup defaultActiveButton="ON" onChange={spy}>
+        <Button name="ON" appearance="secondary">
           ON
         </Button>
-        <Button theme={Theme} name="OFF" appearance="secondary" onClick={spy2}>
+        <Button name="OFF" appearance="secondary" onClick={spy2}>
           OFF
         </Button>
       </ButtonGroup>,
     );
 
-    render
-      .find(Button)
-      .last()
-      .simulate('click');
+    const offButton = getByText('OFF');
+
+    fireEvent.click(offButton);
 
     expect(spy).toHaveBeenCalledTimes(1);
     expect(spy2).toHaveBeenCalledTimes(1);
   });
 
-  it('should throw an error if child has not name props', () => {
-    let error;
-    try {
-      const render = mount(
-        <ButtonGroup theme={Theme} defaultActiveButton="ON">
-          <Button theme={Theme} appearance="secondary">
-            ON
-          </Button>
-          <Button theme={Theme} appearance="secondary">
-            OFF
-          </Button>
+  test('should throw an error if child has not name props', () => {
+    const renderComponent = () =>
+      render(
+        <ButtonGroup defaultActiveButtonName="ON">
+          <Button appearance="secondary">ON</Button>
+          <Button appearance="secondary">OFF</Button>
         </ButtonGroup>,
       );
-      expect(render).toMatchSnapshot();
-    } catch (e) {
-      error = e;
-    }
-
-    expect(error).toBeInstanceOf(Error);
+    expect(renderComponent).toThrow(
```

Comment:
Is that a good point to check if error message is exactly the one you've passed here ? What if something change ? you'll have to change the expected message. You should just check if an error message is throwed or if the response is type `Error`

Reply:
@AntoninMarchardDev  The message will never change (is part of react api)